sqlmesh/core/context.py

```diff
@@ -826,6 +826,88 @@ def plan(
         Returns:
             The populated Plan object.
         """
+        plan_builder = self.plan_builder(
+            environment,
+            start=start,
+            end=end,
+            execution_time=execution_time,
+            create_from=create_from,
+            skip_tests=skip_tests,
+            restate_models=restate_models,
+            no_gaps=no_gaps,
+            skip_backfill=skip_backfill,
+            forward_only=forward_only,
+            no_auto_categorization=no_auto_categorization,
+            effective_from=effective_from,
+            include_unmodified=include_unmodified,
+            select_models=select_models,
+            backfill_models=backfill_models,
+            categorizer_config=categorizer_config,
+            run=run,
+        )
+
+        self.console.plan(
+            plan_builder, auto_apply, self.default_catalog, no_diff=no_diff, no_prompts=no_prompts
+        )
+
+        return plan_builder.build()
+
+    def plan_builder(
```

Comment:
Sure you want to make this public? Not sure if we want to commit to exposing this.

Reply:
Discussed in person. We concluded that this method makes a lot more sense when using SQLMesh API as opposed to the `plan` one which is clearly geared towards using in CLI or Notebook.